analyzers/tests/SonarAnalyzer.Test/Rules/PasswordsShouldBeStoredCorrectlyTest.cs

```diff
@@ -60,28 +90,44 @@ void Method(int iterations, byte[] bs)
                     //  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                         new Rfc2898DeriveBytes("password", bs, 42);                         // Noncompliant
                         new Rfc2898DeriveBytes("password", 42, 42);                         // Noncompliant
-                        new Rfc2898DeriveBytes(bs, bs, 42, HashAlgorithmName.MD5);          // Noncompliant {{Use at least 100,000 iterations here.}}
+                        new Rfc2898DeriveBytes(bs, bs, 42, han);                            // Noncompliant {{Use at least 100,000 iterations here.}}
                     //                                 ^^
-                        new Rfc2898DeriveBytes("password", bs, 42, HashAlgorithmName.MD5);  // Noncompliant
-                        new Rfc2898DeriveBytes("password", 42, 42, HashAlgorithmName.MD5);  // Noncompliant
+                        new Rfc2898DeriveBytes("password", bs, 42, han);                    // Noncompliant
+                        new Rfc2898DeriveBytes("password", 42, 42, han);                    // Noncompliant
                         new Rfc2898DeriveBytes(bs, bs, ITERATIONS);                         // Noncompliant
                         new Rfc2898DeriveBytes("", bs, ITERATIONS);                         // Noncompliant
                         new Rfc2898DeriveBytes("", 42, ITERATIONS);                         // Noncompliant
                     //  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-                        new Rfc2898DeriveBytes(bs, bs, iterations, HashAlgorithmName.SHA1); // Compliant
-                        new Rfc2898DeriveBytes(bs, bs, ITERATIONS, HashAlgorithmName.SHA1); // Compliant
-                        new Rfc2898DeriveBytes("", bs, ITERATIONS, HashAlgorithmName.SHA1); // Compliant
-                        new Rfc2898DeriveBytes("", 42, ITERATIONS, HashAlgorithmName.SHA1); // Compliant
+                        new Rfc2898DeriveBytes(bs, bs, iterations, han);                    // Compliant
+                        new Rfc2898DeriveBytes(bs, bs, ITERATIONS, han);                    // Compliant
+                        new Rfc2898DeriveBytes("", bs, ITERATIONS, han);                    // Compliant
+                        new Rfc2898DeriveBytes("", 42, ITERATIONS, han);                    // Compliant
+
+                        var x = new Rfc2898DeriveBytes(bs, bs, ITERATIONS, han);
+                        x.IterationCount = 1;                                               // Noncompliant {{Use at least 100,000 iterations here.}}
+                    //  ^^^^^^^^^^^^^^^^
+
+                        new Rfc2898DeriveBytes(bs, bs, ITERATIONS, han)
+                        {
+                            IterationCount = 42                                             // Noncompliant {{Use at least 100,000 iterations here.}}
+                    //      ^^^^^^^^^^^^^^
```

Comment:
Even though the constructor is invoked with `ITERATIONS`, this bad boy is re-writting the value to 42.